tools/network/cloudflare/cloudflare.go

```diff
@@ -295,6 +301,37 @@ func (c *Cred) GetZones(ctx context.Context) (zones []Zone, err error) {
 	return zones, err
 }
 
+// GetZoneID returns a zoneID that matches the requested zoneDomainName.
+func (c *Cred) GetZoneID(ctx context.Context, zoneDomainName string) (zoneID string, err error) {
+	zoneID = ""
+	zones, err := c.GetZones(ctx)
+	if err != nil {
+		return
+	}
+	if len(zones) == 0 {
+		err = ErrUserNotPermitted
+		return
+	}
+	zoneDomainName = strings.ToLower(zoneDomainName)
+	var matchingZone *Zone
+	for _, zone := range zones {
+		if zoneDomainName == strings.ToLower(zone.DomainName) {
+			// found a match.
+			if matchingZone != nil {
+				// we already had a previous match ?!
+				err = ErrInvalidDomainName
```

Comment:
Should this error be more explicit?  If we're failing because it's duplicate, shouldn't that be reflected in the err msg?

Reply:
I'm not sure if this is a true duplicate name or not. But I know it's a use case that I've never ran into, and I'd like to build some "counter-measures" if that happens.
If you have any particular error name, please do share.
( would ErrDuplicateZoneNameFound sounds any better ? )

Reply:
sounds better and more importantly the error text should reflect that too

Reply:
sure. done.